src/VisualStudio/Core/Def/Storage/VisualStudioCloudCacheStorageService.cs

```diff
@@ -17,36 +17,25 @@ namespace Microsoft.VisualStudio.LanguageServices.Storage
     internal class VisualStudioCloudCacheStorageService : AbstractCloudCachePersistentStorageService
     {
         private readonly IAsyncServiceProvider _serviceProvider;
-        private readonly IThreadingContext _threadingContext;
 
-        public VisualStudioCloudCacheStorageService(IAsyncServiceProvider serviceProvider, IThreadingContext threadingContext, IPersistentStorageConfiguration configuration)
+        public VisualStudioCloudCacheStorageService(IAsyncServiceProvider serviceProvider, IPersistentStorageConfiguration configuration)
             : base(configuration)
         {
             _serviceProvider = serviceProvider;
-            _threadingContext = threadingContext;
         }
 
-        protected sealed override void DisposeCacheService(ICacheService cacheService)
-        {
-            if (cacheService is IAsyncDisposable asyncDisposable)
-            {
-                _threadingContext.JoinableTaskFactory.Run(
-                    () => asyncDisposable.DisposeAsync().AsTask());
-            }
-            else if (cacheService is IDisposable disposable)
-            {
-                disposable.Dispose();
-            }
-        }
-
-        protected sealed override async ValueTask<ICacheService> CreateCacheServiceAsync(CancellationToken cancellationToken)
+        protected sealed override async ValueTask<ICacheService> CreateCacheServiceAsync(string solutionFolder, CancellationToken cancellationToken)
         {
             var serviceContainer = await _serviceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>().ConfigureAwait(false);
             var serviceBroker = serviceContainer.GetFullAccessServiceBroker();
 
 #pragma warning disable ISB001 // Dispose of proxies
             // cache service will be disposed inside VisualStudioCloudCachePersistentStorage.Dispose
-            var cacheService = await serviceBroker.GetProxyAsync<ICacheService>(VisualStudioServices.VS2019_10.CacheService, cancellationToken: cancellationToken).ConfigureAwait(false);
+            var cacheService = await serviceBroker.GetProxyAsync<ICacheService>(
+                VisualStudioServices.VS2019_10.CacheService,
+                // replace with CacheService.RelativePathBaseActivationArgKey once available.
```

Comment:
I don't think you reference the cache service package, do you? Everything you use I think you get from VS.RPC.Contracts.  I don't know that we'll ever declare a constant for this there. At least, there's no great place for it.

Reply:
> At least, there's no great place for it.

Can't it be in a type exported from teh same assembly that exports ICacheService?  You could also just make it a constant inside the ICacheService interface (though i don't love that).

Reply:
Interfaces can't declare constants, last I checked. 
The assembly that exports `ICacheService` is the VS.RPC.Contracts assembly that I mentioned above. We would need to declare the constant on some other class in that assembly.

Reply:
Interfaces can declare constants. :-)

Reply:
Cool. But in trying this out, C# 8 is required, and C# forbids constants on interfaces except when targeting .NET Core 3.1 or later. VS.RPC.Contracts targets netstandard2.0 as net472 is still of course an important target.

Reply:
Are you guys not on C#8? :)   I didn't know about teh netstandard issue.  That certainly seems weird as netstandard should support this.  Bummer.

Reply:
Yes, we absolutely leverage C# 8 -- C# 9, in fact. I don't mind using syntax that requires C# 9. But ya, your compiler forbids constants on interfaces except when targeting a runtime that supports the default method implementations feature.